service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java

```diff
@@ -685,4 +689,246 @@ public void changeTariffLocationsStatus(Long tariffId, ChangeTariffLocationStatu
+     *                courier id.
+     * @return true if you have to deactivate tariff by receiving stations and false
+     *         if not.
+     * @author Nikita Korzh.
+     */
+    private boolean shouldDeactivateTariffsByReceivingStations(DetailsOfDeactivateTariffsDto details) {
+        if (details.getStationsId().isPresent() && details.getRegionsId().isEmpty()
+            && details.getCitiesId().isEmpty() && details.getCourierId().isEmpty()) {
+            if (deactivateTariffsForChosenParamRepository
+                .checkIfReceivingStationsExists(details.getStationsId().get())) {
+                return true;
+            } else {
+                throw new NotFoundException("Current receiving stations don't exists");
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Method that checks if the tariff should be deactivated by courier id.
+     *
+     * @param details - contains regions id, cities id, receiving stations id and
+     *                courier id.
+     * @return true if you have to deactivate tariff by courier id and false if not.
+     * @author Nikita Korzh.
+     */
+    private boolean shouldDeactivateTariffsByCourier(DetailsOfDeactivateTariffsDto details) {
+        if (details.getCourierId().isPresent() && details.getRegionsId().isEmpty()
+            && details.getCitiesId().isEmpty() && details.getStationsId().isEmpty()) {
+            if (courierRepository.existsCourierById(details.getCourierId().get())) {
+                return true;
+            } else {
+                throw new NotFoundException("Current courier doesn't exists");
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Method that checks if the tariff should be deactivated by region id and
+     * cities id.
+     *
+     * @param details - contains regions id, cities id, receiving stations id and
+     *                courier id.
+     * @return true if you have to deactivate tariff by region id and cities id and
+     *         false if not.
+     * @author Nikita Korzh.
+     */
+    private boolean shouldDeactivateTariffsByRegionsAndCities(DetailsOfDeactivateTariffsDto details) {
+        if (details.getRegionsId().isPresent() && details.getCitiesId().isPresent()
+            && details.getStationsId().isEmpty() && details.getCourierId().isEmpty()) {
+            if (details.getRegionsId().get().size() == 1) {
+                if (regionRepository.existsRegionById(details.getRegionsId().get().get(0))
+                    && deactivateTariffsForChosenParamRepository.checkIfCitiesExistForRegion(
+                        details.getCitiesId().get(),
+                        details.getRegionsId().get().get(0))) {
+                    return true;
+                } else {
+                    throw new NotFoundException("Current region or cities don't exists");
+                }
+            } else {
+                throw new BadRequestException("Size of regions id should be 1");
```

Comment:
`/deactivate `endpoint takes a list of `regionIds`, please add info on why the size should be 1 in this case